firewood/src/merkle.rs

```diff
@@ -355,58 +343,73 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
         val: Vec<u8>,
         root: DiskAddress,
     ) -> Result<(), MerkleError> {
+        let (parents, deleted) = self.insert_and_return_updates(key, val, root)?;
+
+        for mut r in parents {
```

Comment:
🏆 
I was thinking: `parents.iter().for_each(|mut &r| r.write(|u| u.rehash())?);` but yours is more readable.

Reply:
You could do it with `iter_mut` and `try_for_each`, but that's a bigger change than what I wanted to do. You can see the original code on the left somewhere below:
```rust
        for (mut r, _) in parents.into_iter().rev() {
            r.write(|u| u.rehash()).unwrap();
        }
```

The reversed iterator is now returned from `insert_and_return_updates` and I drop the indexes as they aren't used here.